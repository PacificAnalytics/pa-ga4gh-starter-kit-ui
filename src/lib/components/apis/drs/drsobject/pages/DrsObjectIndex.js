import '@fontsource/roboto';
import React, { useEffect } from 'react';
import { 
  Typography, 
  Container, 
  Table,
  TableContainer, 
  TableHead, 
  TableBody, 
  TableRow, 
  TableCell, 
  Button, 
  Grid,
  Paper,
  Breadcrumbs,
  Link as MuiLink
} from '@material-ui/core';
import ArrowBackIos from '@material-ui/icons/ArrowBackIos';
import {
  Link, 
  useLocation
} from "react-router-dom";
import { scrollToTop } from '../../../../functions/common';
import PageContainer from '../../../../common/layout/PageContainer';

  /* Render index table rows populated with data */
const DrsIndexRows = (props) => {
  const drsObjectsList = props.drsObjectsList;
  if (!drsObjectsList){
    return null;
  }
  else {
    const rows = drsObjectsList.map((drsObject) =>
    <TableRow key={drsObject.id}>
      <TableCell align="left">{drsObject.id}</TableCell>
      <TableCell align="left">{drsObject.name}</TableCell>
      <TableCell align="left">
        <Button variant="outlined" color='secondary' component={Link} to={`/drs/${drsObject.id}`}>
          View
        </Button>
      </TableCell>
      <TableCell align="center">
        <IconButton 
          variant="contained" 
          color='primary' 
          component={Link} 
          to={`/drs/${drsObject.id}/edit`}
          onClick={scrollToTop}
        >
          <EditIcon/>
        </IconButton>
      </TableCell>
    </TableRow>
    );
    return (
      <TableBody>{rows}</TableBody>
    );
  }
}

const DrsObjectIndex = (props) => {
  /* Restore scroll to top of page on navigation to a new page */
  console.log('inside DRS Object Index!!');
  const { pathname }  = useLocation();
  useEffect(() => {
    window.scrollTo(0, 0);
  }, [pathname])

  /* Render DrsIndex page */
  return (
    <PageContainer>
      <Breadcrumbs>
        <Link color="inherit" to="/home">starter-kit</Link>
        <Link>services</Link>
        <Link>org.ga4gh.my-drs-service</Link>
        <Link to="/drs">drs</Link>
      </Breadcrumbs>

      <div>
        <Button
          variant='text'
          component={Link}
          to='/'
          color='primary'
        >
          <ArrowBackIos />
          <Typography>back</Typography>
        </Button>
      </div>
      <div>
        <Button variant='contained' component={Link} to='/drs/new' color='primary' size='large'>
          <Typography variant='button'>New DRS Object</Typography>
        </Button>
      </div>
    
      <TableContainer component={Paper}>
        <Table size="small">
          <TableHead>
            <TableRow>
              <TableCell align="left">DRS Object ID</TableCell>
              <TableCell align="left">Name</TableCell>
              <TableCell align="left" />
            </TableRow>
          </TableHead>
          <DrsIndexRows drsObjectsList={props.drsObjectsList}/>
        </Table>
      </TableContainer>
    </PageContainer>
  );
}

export default DrsObjectIndex;
