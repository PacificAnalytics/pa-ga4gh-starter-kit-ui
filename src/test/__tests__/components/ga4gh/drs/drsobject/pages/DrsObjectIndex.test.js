import React from 'react';
import renderer from 'react-test-renderer';
import {render, screen, getByRole, queryByRole, getByLabelText, waitFor, logRoles, getByText} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import {
    mockDrsObjectsList,
    mockDrsObjectMainTrail
} from '../../../../../../resources/MockData';
import DrsObjectIndex from '../../../../../../../lib/components/ga4gh/drs/drsobject/pages/DrsObjectIndex';
import { MemoryRouter } from 'react-router';

test('Render <DrsObjectIndex /> component with data', () => {
    let {container} = render(
        <MemoryRouter>
            <DrsObjectIndex
                trail={mockDrsObjectMainTrail}
                drsObjectsList={mockDrsObjectsList}
            />
        </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByText('DRS Object ID')).toBeInTheDocument();
    expect(screen.getByText('Name')).toBeInTheDocument();

    mockDrsObjectsList.forEach((drsObject) => {
        let indexRow = screen.getByRole('row', {name: drsObject.id})
        expect(getByText(indexRow, `${drsObject.id}`)).toBeInTheDocument();
        expect(getByText(indexRow, `${drsObject.name}`)).toBeInTheDocument();
        expect(getByLabelText(indexRow, 'view-button')).toBeInTheDocument();
        expect(getByLabelText(indexRow, 'edit-button')).toBeInTheDocument();
    });
});

test('Render <DrsObjectIndex /> component without data', () => {
    let {container} = render(
        <MemoryRouter>
            <DrsObjectIndex
                trail={mockDrsObjectMainTrail}
                drsObjectsList={[]}
            />
        </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByText('DRS Object ID')).toBeInTheDocument();
    expect(screen.getByText('Name')).toBeInTheDocument();
    expect(screen.queryByLabelText('view-button')).not.toBeInTheDocument();
    expect(screen.queryByLabelText('edit-button')).not.toBeInTheDocument();
});