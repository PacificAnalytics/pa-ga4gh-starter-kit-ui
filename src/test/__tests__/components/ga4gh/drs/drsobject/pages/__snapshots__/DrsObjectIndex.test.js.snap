// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <DrsObjectIndex /> component with data 1`] = `
<div
  class="MuiContainer-root makeStyles-root-2 MuiContainer-maxWidthLg"
>
  <nav
    class="MuiTypography-root MuiBreadcrumbs-root makeStyles-root-3 MuiTypography-body1 MuiTypography-colorTextSecondary"
  >
    <ol
      class="MuiBreadcrumbs-ol"
    >
      <li
        class="MuiBreadcrumbs-li"
      >
        <span>
          <a
            class="makeStyles-link-4"
            color="inherit"
            href="/home"
          >
            starter-kit
          </a>
        </span>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        /
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <span>
          <a
            class="makeStyles-link-4"
            color="inherit"
            href="/services"
          >
            services
          </a>
        </span>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        /
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <span>
          <a
            class="makeStyles-link-4"
            color="inherit"
            href="/services/org.ga4gh.starterkit.drs"
          >
            org.ga4gh.starterkit.drs
          </a>
        </span>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        /
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <span>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            drs
          </p>
        </span>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        /
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <span>
          <a
            class="makeStyles-link-4"
            color="inherit"
            href="/services/org.ga4gh.starterkit.drs/drs/objects"
          >
            objects
          </a>
        </span>
      </li>
    </ol>
  </nav>
  <a
    aria-disabled="false"
    aria-label="back-button"
    class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-root-5 MuiButton-textPrimary"
    href="/"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
        />
      </svg>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        back
      </p>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </a>
  <div>
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-newButton-1 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
      href="/undefined/new"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiTypography-root MuiTypography-button"
        >
          New DRS Object
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </a>
  </div>
  <div
    class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      class="MuiTable-root"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall"
            scope="col"
          >
            DRS Object ID
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall"
            scope="col"
          >
            Name
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall"
            scope="col"
          />
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          aria-label="b8cd0667-2c33-4c9f-967b-161b905932c9"
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall"
          >
            b8cd0667-2c33-4c9f-967b-161b905932c9
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall"
          >
            phenopackets.test.dataset
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall"
          >
            <a
              aria-disabled="false"
              aria-label="view-button"
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-6 MuiButton-outlinedSecondary"
              href="/undefined/b8cd0667-2c33-4c9f-967b-161b905932c9"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                View
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <a
              aria-disabled="false"
              aria-label="edit-button"
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-7"
              href="/undefined/b8cd0667-2c33-4c9f-967b-161b905932c9/edit"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                Edit
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <div
              class="makeStyles-div-8"
            >
              <button
                aria-label="delete-button"
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-9"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Delete
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </td>
        </tr>
        <tr
          aria-label="a1dd4ae2-8d26-43b0-a199-342b64c7dff6"
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall"
          >
            a1dd4ae2-8d26-43b0-a199-342b64c7dff6
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall"
          >
            phenopackets.cao.family
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall"
          >
            <a
              aria-disabled="false"
              aria-label="view-button"
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-6 MuiButton-outlinedSecondary"
              href="/undefined/a1dd4ae2-8d26-43b0-a199-342b64c7dff6"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                View
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <a
              aria-disabled="false"
              aria-label="edit-button"
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-7"
              href="/undefined/a1dd4ae2-8d26-43b0-a199-342b64c7dff6/edit"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                Edit
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <div
              class="makeStyles-div-8"
            >
              <button
                aria-label="delete-button"
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-9"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Delete
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </td>
        </tr>
        <tr
          aria-label="1a570e4e-2489-4218-9333-f65549495872"
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall"
          >
            1a570e4e-2489-4218-9333-f65549495872
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall"
          >
            phenopackets.cao.1
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall"
          >
            <a
              aria-disabled="false"
              aria-label="view-button"
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-6 MuiButton-outlinedSecondary"
              href="/undefined/1a570e4e-2489-4218-9333-f65549495872"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                View
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <a
              aria-disabled="false"
              aria-label="edit-button"
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-7"
              href="/undefined/1a570e4e-2489-4218-9333-f65549495872/edit"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                Edit
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <div
              class="makeStyles-div-8"
            >
              <button
                aria-label="delete-button"
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-9"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Delete
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </td>
        </tr>
        <tr
          aria-label="4d83ba3f-a476-4c7c-868f-3d1fcf77fe29"
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall"
          >
            4d83ba3f-a476-4c7c-868f-3d1fcf77fe29
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall"
          >
            phenopackets.cao.2
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall"
          >
            <a
              aria-disabled="false"
              aria-label="view-button"
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-6 MuiButton-outlinedSecondary"
              href="/undefined/4d83ba3f-a476-4c7c-868f-3d1fcf77fe29"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                View
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <a
              aria-disabled="false"
              aria-label="edit-button"
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-7"
              href="/undefined/4d83ba3f-a476-4c7c-868f-3d1fcf77fe29/edit"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                Edit
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <div
              class="makeStyles-div-8"
            >
              <button
                aria-label="delete-button"
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-9"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Delete
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </td>
        </tr>
        <tr
          aria-label="924901d5-6d31-4c33-b443-7931eadfac4b"
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall"
          >
            924901d5-6d31-4c33-b443-7931eadfac4b
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall"
          >
            phenopackets.cao.3
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall"
          >
            <a
              aria-disabled="false"
              aria-label="view-button"
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-6 MuiButton-outlinedSecondary"
              href="/undefined/924901d5-6d31-4c33-b443-7931eadfac4b"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                View
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <a
              aria-disabled="false"
              aria-label="edit-button"
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-7"
              href="/undefined/924901d5-6d31-4c33-b443-7931eadfac4b/edit"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                Edit
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <div
              class="makeStyles-div-8"
            >
              <button
                aria-label="delete-button"
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-9"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Delete
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </td>
        </tr>
        <tr
          aria-label="0f8abce3-e161-4bdf-981f-86257d505d69"
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall"
          >
            0f8abce3-e161-4bdf-981f-86257d505d69
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall"
          >
            phenopackets.cao.4
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall"
          >
            <a
              aria-disabled="false"
              aria-label="view-button"
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-6 MuiButton-outlinedSecondary"
              href="/undefined/0f8abce3-e161-4bdf-981f-86257d505d69"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                View
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <a
              aria-disabled="false"
              aria-label="edit-button"
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-7"
              href="/undefined/0f8abce3-e161-4bdf-981f-86257d505d69/edit"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                Edit
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <div
              class="makeStyles-div-8"
            >
              <button
                aria-label="delete-button"
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-9"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Delete
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Render <DrsObjectIndex /> component without data 1`] = `
<div
  class="MuiContainer-root makeStyles-root-11 MuiContainer-maxWidthLg"
>
  <nav
    class="MuiTypography-root MuiBreadcrumbs-root makeStyles-root-12 MuiTypography-body1 MuiTypography-colorTextSecondary"
  >
    <ol
      class="MuiBreadcrumbs-ol"
    >
      <li
        class="MuiBreadcrumbs-li"
      >
        <span>
          <a
            class="makeStyles-link-13"
            color="inherit"
            href="/home"
          >
            starter-kit
          </a>
        </span>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        /
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <span>
          <a
            class="makeStyles-link-13"
            color="inherit"
            href="/services"
          >
            services
          </a>
        </span>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        /
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <span>
          <a
            class="makeStyles-link-13"
            color="inherit"
            href="/services/org.ga4gh.starterkit.drs"
          >
            org.ga4gh.starterkit.drs
          </a>
        </span>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        /
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <span>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            drs
          </p>
        </span>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        /
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <span>
          <a
            class="makeStyles-link-13"
            color="inherit"
            href="/services/org.ga4gh.starterkit.drs/drs/objects"
          >
            objects
          </a>
        </span>
      </li>
    </ol>
  </nav>
  <a
    aria-disabled="false"
    aria-label="back-button"
    class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-root-14 MuiButton-textPrimary"
    href="/"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
        />
      </svg>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        back
      </p>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </a>
  <div>
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-newButton-10 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
      href="/undefined/new"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiTypography-root MuiTypography-button"
        >
          New DRS Object
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </a>
  </div>
  <div
    class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      class="MuiTable-root"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall"
            scope="col"
          >
            DRS Object ID
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall"
            scope="col"
          >
            Name
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall"
            scope="col"
          />
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      />
    </table>
  </div>
</div>
`;
